### PR TITLE
Menu option save config

### PR DESCRIPTION
**Pull request by svartkanin**
Addressing #1002

Adding a new menu option to save the current configuration to file. It allows for saving individual configurations or all of them.

save_option

In addition, I've also included a preview for missing configurations to start the installation process (in case people might not find it obvious what's missing)

install_output

### CODE REVIEW

**Comment by wllacer** [comment]
Wow !!!
Have a look at the changes for menu.py at my PR #1005.  There i decided to stop adding more parameters to Menu and use **kwargs for new additions to be passed to TerminalMenu (simple_menu.py), so there is no need for you to handle at Menu level the preview* parameters

File: archinstall/lib/menu/selection_menu.py [hunk 1]
```diff
@@ -544,31 +582,37 @@ def exit_callback(self):
...
-			missing += 1
-		if not check('timezone'):
-			missing += 1
+			missing += ['Audio']
 		if not check('!root-password') and not check('!superusers'):
-			missing += 1
+			missing += [str(_('Either root-password or at least 1 superuser must be specified'))]
 		if not check('harddrives'):
-			missing += 1
+			missing += ['Harddrives']
```
Comment (dylanmtaylor) [inline archinstall/lib/menu/selection_menu.py:612]: Nit pick: This shouldn't be one word.

Comment (dylanmtaylor) [reply archinstall/lib/menu/selection_menu.py:612]: Or, alternatively it could just be 'Drives'.

Comment (Torxed) [reply archinstall/lib/menu/selection_menu.py:612]: Might need to be changed elsewhere too if we do. Visually it probably doesn't matter if we only change here. But the heck for `if .."harddrives"` has to be changed to `if .."hard drives"` as well where we rely on this keyword :)

**Comment by Torxed** [comment]
Managed to peace together the parameter changes I think :)

**Comment by svartkanin** [comment]
> Have a look at the changes for menu.py at my PR #1005. There i decided to stop adding more parameters to Menu and use **kwargs for new additions to be passed to TerminalMenu (simple_menu.py), so there is no need for you to handle at Menu level the preview* parameters

@wllacer yes I saw that but I had some concerns with that (i think I left a comment on your pr). I think kwargs hides information and it makes it difficult to search for usages of parameters throughout the code. I get that the parameters here are quite extensive, however, I think the benefits are still higher in keeping it explicit :)

**Comment by Torxed** [comment]
Yea I stopped using `**kwargs` only because it's ambiguous.
It also makes it hard for code linting and stuff, as the list of supported objects that `kwargs` can take can grow indefinitely.

I like the method tho, makes the init look a lot cleaner ^^

**Pull request merged**